src/main/java/org/truffleruby/core/module/ModuleFields.java

```diff
@@ -87,7 +87,11 @@ public static void debugModuleChain(DynamicObject module) {
     private final ConcurrentMap<String, InternalMethod> methods = new ConcurrentHashMap<>();
     private final ConcurrentMap<String, RubyConstant> constants = new ConcurrentHashMap<>();
     private final ConcurrentMap<String, Object> classVariables = new ConcurrentHashMap<>();
+
+    // Maps a refined class to refinement module
```

Comment:
Ah, so this is a list of refinements within a module wrapping `refine` calls, like `B` in your example.
Could you try to clarify?

Reply:
This is where a modules refinements are actually stored so that when you call `using Mod` later they will be looked up here and added to the scope.  I've added an extended description of refinments here:
https://github.com/graalvm/truffleruby/pull/860#issuecomment-352157596